index.js

```diff
@@ -418,6 +417,10 @@ function browserslist(queries, opts) {
     }
   }
 
+  if (hasPathAwareQueries(queries)) {
+    context.path = opts.path;
```

Comment:
This fix is too hacky. 1 year later I will forget this hack and will spend a night trying to find why `context` is different for some cases.

We should keep `context` the same, but change cache algorithm instead.

Reply:
Are you suggesting removing path during cacheKey generation? I thought it would be more risky since accessing context.path from another query would appear to work initially, but fail subtly due to caching, as opposed to failing loudly due to missing path in the current fix.

The clean solution would probably be: 
1. Add explicit dependencies, eg ['path', 'mobileToDesktop'] to every query
2. Do a parse() pass on query list to split on logical operators
3. Form context with discovered dependencies
But it would be less efficient due to parsing, and probably not that useful as the query set is probably stable by now. 

If you have some entirely different caching algorithm in mind, I'd be glad to add it, but I don't see what it could be.

Reply:
I suggest to *not* remove `path` from `context` and instead change the code which use cache.

> The clean solution would probably be:

Also, a way but maybe too complex.

Can you show me the exact place where you want to change cache logic?

Reply:
If we attack from a different angle, we can normalize `path` to reduce cardinality — even path-dependent queries don't seem to need the precise current file path:

- `browserslist_config` needs the closest directory with browserslist config
- `extends` needs the closest `package.json` (unless using `dangerousExtend`)

So we could replace `path` with something like

```js
var context = {
  extendsRoot: dangerousExtend ? opts.path : findClosestPackageJson(opts.path),
  parentConfig: env.findConfigFile(opts.path),
  ...
};
```

In this case, normalizing the path involves FS, which brings us back to #859

Reply:
Hm, interesting.

Let’s finish with #864 and then will try to do this.

Reply:
before we go all crazy — pls take a look at the updated option, is this too hacky still?